### ISSUE TITLE
感觉huggingface transformers的模型参数未加载成功？

### ISSUE THREAD

**Issue by huangyc0618** [bytedance/lightseq#135]
跑了几个分类任务，前面的epoch acc都是0.5，和直接跑huggingface transformer比相差很大

**Comment by Taka152** [comment]
Could you check your lightseq version? Fully supported BertEncoder is
released in  2.1.0.

**Comment by huangyc0618** [comment]
lightseq version is 2.1.0.

**Comment by huangyc0618** [comment]
example里的run_glue，用了lightseq的loss变化：
```
{'loss': 9.5076, 'learning_rate': 1.9851851851851855e-05, 'epoch': 0.06}                                                                                    
{'loss': 2.8418, 'learning_rate': 1.9481481481481484e-05, 'epoch': 0.11}                                                                                    
{'loss': 2.28, 'learning_rate': 1.9111111111111113e-05, 'epoch': 0.17}                                                                                      
{'loss': 2.1608, 'learning_rate': 1.8740740740740742e-05, 'epoch': 0.22}                                                                                    
{'loss': 2.0418, 'learning_rate': 1.837037037037037e-05, 'epoch': 0.28}                                                                                     
{'loss': 2.2944, 'learning_rate': 1.8e-05, 'epoch': 0.33}                                                                                                   
{'loss': 2.2678, 'learning_rate': 1.7629629629629633e-05, 'epoch': 0.39}                                                                                    
{'loss': 2.1319, 'learning_rate': 1.725925925925926e-05, 'epoch': 0.44}                                                                                     
{'loss': 1.9258, 'learning_rate': 1.688888888888889e-05, 'epoch': 0.5}                                                                                      
{'loss': 1.9355, 'learning_rate': 1.651851851851852e-05, 'epoch': 0.56}                                                                                     
{'loss': 2.0799, 'learning_rate': 1.614814814814815e-05, 'epoch': 0.61}                                                                                     
{'loss': 2.0739, 'learning_rate': 1.577777777777778e-05, 'epoch': 0.67}                                                                                     
{'loss': 2.147, 'learning_rate': 1.5407407407407408e-05, 'epoch': 0.72}                                                                                     
{'loss': 1.9541, 'learning_rate': 1.5037037037037039e-05, 'epoch': 0.78}                                                                                    
{'loss': 2.2207, 'learning_rate': 1.4666666666666666e-05, 'epoch': 0.83}                                                                                    
{'loss': 2.0572, 'learning_rate': 1.4296296296296297e-05, 'epoch': 0.89}                                                                                    
{'loss': 2.026, 'learning_rate': 1.3925925925925928e-05, 'epoch': 0.94}                                                                                     
{'loss': 1.7824, 'learning_rate': 1.3555555555555557e-05, 'epoch': 1.0}                                                                                     
{'loss': 1.8262, 'learning_rate': 1.3185185185185185e-05, 'epoch': 1.06}                                                                                    
{'loss': 1.6327, 'learning_rate': 1.2814814814814815e-05, 'epoch': 1.11}                                                                                    
{'loss': 1.6313, 'learning_rate': 1.2444444444444446e-05, 'epoch': 1.17}                                                                                    
{'loss': 1.547, 'learning_rate': 1.2074074074074075e-05, 'epoch': 1.22}                                                                                     
{'loss': 1.4633, 'learning_rate': 1.1703703703703703e-05, 'epoch': 1.28}                                                                                    
{'loss': 1.4194, 'learning_rate': 1.1333333333333334e-05, 'epoch': 1.33}                                                                                    
{'loss': 1.2543, 'learning_rate': 1.0962962962962965e-05, 'epoch': 1.39}                                                                                    
{'loss': 1.389, 'learning_rate': 1.0592592592592594e-05, 'epoch': 1.44}                                                                                     
{'loss': 1.3184, 'learning_rate': 1.0222222222222223e-05, 'epoch': 1.5}                                                                                     
{'loss': 1.2084, 'learning_rate': 9.851851851851852e-06, 'epoch': 1.56}                                                                                     
{'loss': 1.3246, 'learning_rate': 9.481481481481483e-06, 'epoch': 1.61}                                                                                     
{'loss': 1.2105, 'learning_rate': 9.111111111111112e-06, 'epoch': 1.67}                                                                                     
{'loss': 1.1819, 'learning_rate': 8.740740740740741e-06, 'epoch': 1.72}                                                                                     
{'loss': 0.9332, 'learning_rate': 8.37037037037037e-06, 'epoch': 1.78}                                                                                      
{'loss': 1.0241, 'learning_rate': 8.000000000000001e-06, 'epoch': 1.83}                                                                                     
{'loss': 1.0881, 'learning_rate': 7.62962962962963e-06, 'epoch': 1.89}                                                                                      
{'loss': 1.0188, 'learning_rate': 7.2592592592592605e-06, 'epoch': 1.94}                                                                                    
{'loss': 1.1178, 'learning_rate': 6.88888888888889e-06, 'epoch': 2.0}                                                                                       
{'loss': 0.9385, 'learning_rate': 6.51851851851852e-06, 'epoch': 2.06}                                                                                      
{'loss': 0.9125, 'learning_rate': 6.148148148148149e-06, 'epoch': 2.11}                                                                                     
{'loss': 0.9146, 'learning_rate': 5.777777777777778e-06, 'epoch': 2.17}                                                                                     
{'loss': 0.9386, 'learning_rate': 5.407407407407408e-06, 'epoch': 2.22}                                                                                     
{'loss': 0.9709, 'learning_rate': 5.037037037037037e-06, 'epoch': 2.28}                                                                                     
{'loss': 0.9208, 'learning_rate': 4.666666666666667e-06, 'epoch': 2.33}                                                                                     
{'loss': 0.9185, 'learning_rate': 4.296296296296296e-06, 'epoch': 2.39}                                                                                     
{'loss': 1.1793, 'learning_rate': 3.925925925925926e-06, 'epoch': 2.44}                                                                                     
{'loss': 0.9275, 'learning_rate': 3.555555555555556e-06, 'epoch': 2.5}                                                                                      
{'loss': 0.8783, 'learning_rate': 3.1851851851851855e-06, 'epoch': 2.56}                                                                                    
{'loss': 0.9011, 'learning_rate': 2.814814814814815e-06, 'epoch': 2.61}                                                                                     
{'loss': 1.0595, 'learning_rate': 2.4444444444444447e-06, 'epoch': 2.67}                                                                                    
{'loss': 0.8952, 'learning_rate': 2.0740740740740742e-06, 'epoch': 2.72}                                                                                    
{'loss': 0.8996, 'learning_rate': 1.7037037037037038e-06, 'epoch': 2.78}  

```

没用lightseq的原生huggingface loss变化：
```
{'loss': 11.3951, 'learning_rate': 1.977777777777778e-05, 'epoch': 0.06}                                                                                    
{'loss': 5.0952, 'learning_rate': 1.9407407407407407e-05, 'epoch': 0.11}                                                                                    
{'loss': 3.0288, 'learning_rate': 1.903703703703704e-05, 'epoch': 0.17}                                                                                     
{'loss': 2.0042, 'learning_rate': 1.866666666666667e-05, 'epoch': 0.22}                                                                                     
{'loss': 1.6484, 'learning_rate': 1.8296296296296298e-05, 'epoch': 0.28}                                                                                    
{'loss': 1.1605, 'learning_rate': 1.7925925925925927e-05, 'epoch': 0.33}                                                                                    
{'loss': 1.0815, 'learning_rate': 1.7555555555555556e-05, 'epoch': 0.39}                                                                                    
{'loss': 0.995, 'learning_rate': 1.7185185185185185e-05, 'epoch': 0.44}                                                                                     
{'loss': 0.9219, 'learning_rate': 1.6814814814814818e-05, 'epoch': 0.5}                                                                                     
{'loss': 0.924, 'learning_rate': 1.6444444444444444e-05, 'epoch': 0.56}                                                                                     
{'loss': 0.8325, 'learning_rate': 1.6074074074074076e-05, 'epoch': 0.61}                                                                                    
{'loss': 0.8563, 'learning_rate': 1.5703703703703705e-05, 'epoch': 0.67}                                                                                    
{'loss': 0.9717, 'learning_rate': 1.5333333333333334e-05, 'epoch': 0.72}                                                                                    
{'loss': 0.7686, 'learning_rate': 1.4962962962962964e-05, 'epoch': 0.78}                                                                                    
{'loss': 0.8447, 'learning_rate': 1.4592592592592594e-05, 'epoch': 0.83}                                                                                    
{'loss': 0.7708, 'learning_rate': 1.4222222222222224e-05, 'epoch': 0.89}                                                                                    
{'loss': 0.7552, 'learning_rate': 1.3851851851851851e-05, 'epoch': 0.94}                                                                                    
{'loss': 0.799, 'learning_rate': 1.3481481481481482e-05, 'epoch': 1.0}                                                                                      
{'loss': 0.5932, 'learning_rate': 1.3111111111111113e-05, 'epoch': 1.06}                                                                                    
{'loss': 0.5712, 'learning_rate': 1.2740740740740742e-05, 'epoch': 1.11}                                                                                    
{'loss': 0.6273, 'learning_rate': 1.2370370370370371e-05, 'epoch': 1.17}                                                                                    
{'loss': 0.6329, 'learning_rate': 1.2e-05, 'epoch': 1.22}                                                                                                   
{'loss': 0.5967, 'learning_rate': 1.1629629629629631e-05, 'epoch': 1.28}                                                                                    
{'loss': 0.6135, 'learning_rate': 1.125925925925926e-05, 'epoch': 1.33}                                                                                     
{'loss': 0.6049, 'learning_rate': 1.088888888888889e-05, 'epoch': 1.39}                                                                                     
{'loss': 0.4761, 'learning_rate': 1.0518518518518519e-05, 'epoch': 1.44}                                                                                    
{'loss': 0.5446, 'learning_rate': 1.014814814814815e-05, 'epoch': 1.5}                                                                                      
{'loss': 0.4397, 'learning_rate': 9.777777777777779e-06, 'epoch': 1.56}                                                                                     
{'loss': 0.5698, 'learning_rate': 9.407407407407408e-06, 'epoch': 1.61}                                                                                     
{'loss': 0.5907, 'learning_rate': 9.037037037037037e-06, 'epoch': 1.67}                                                                                     
{'loss': 0.462, 'learning_rate': 8.666666666666668e-06, 'epoch': 1.72}                                                                                      
{'loss': 0.4307, 'learning_rate': 8.296296296296297e-06, 'epoch': 1.78}                                                                                     
{'loss': 0.5116, 'learning_rate': 7.925925925925926e-06, 'epoch': 1.83}                                                                                     
{'loss': 0.5366, 'learning_rate': 7.555555555555556e-06, 'epoch': 1.89}                                                                                     
{'loss': 0.484, 'learning_rate': 7.185185185185186e-06, 'epoch': 1.94}                                                                                      
{'loss': 0.464, 'learning_rate': 6.814814814814815e-06, 'epoch': 2.0}                                                                                       
{'loss': 0.3974, 'learning_rate': 6.444444444444445e-06, 'epoch': 2.06}                                                                                     
{'loss': 0.3406, 'learning_rate': 6.0740740740740745e-06, 'epoch': 2.11}                                                                                    
{'loss': 0.3874, 'learning_rate': 5.7037037037037045e-06, 'epoch': 2.17}                                                                                    
{'loss': 0.3713, 'learning_rate': 5.333333333333334e-06, 'epoch': 2.22}                                                                                     
{'loss': 0.4183, 'learning_rate': 4.962962962962964e-06, 'epoch': 2.28}                                                                                     
{'loss': 0.3988, 'learning_rate': 4.592592592592593e-06, 'epoch': 2.33}                                                                                     
{'loss': 0.3863, 'learning_rate': 4.222222222222223e-06, 'epoch': 2.39}                                                                                     
{'loss': 0.3951, 'learning_rate': 3.851851851851852e-06, 'epoch': 2.44}                                                                                     
{'loss': 0.3862, 'learning_rate': 3.481481481481482e-06, 'epoch': 2.5}                                                                                      
{'loss': 0.3212, 'learning_rate': 3.1111111111111116e-06, 'epoch': 2.56}                                                                                    
{'loss': 0.3836, 'learning_rate': 2.740740740740741e-06, 'epoch': 2.61}                                                                                     
{'loss': 0.4438, 'learning_rate': 2.3703703703703707e-06, 'epoch': 2.67}                                                                                    
{'loss': 0.3413, 'learning_rate': 2.0000000000000003e-06, 'epoch': 2.72}                                                                                    
{'loss': 0.3277, 'learning_rate': 1.62962962962963e-06, 'epoch': 2.78}  
```

两者差的还是非常多的

**Comment by Taka152** [comment]
Thanks for your feedback, we found lightseq on huggingface finetuning is not stable may because of the randomization in training, in some settings of `--seed`, the loss converge as huggingface, but in others, lightseq will stop at some local optimal loss.

So we also do some pre-training experiments on lightseq Bert, and it works fine. We may find another time to fix finetuning instability.
